contracts/src/registry/MetadataMixin.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "./IRegistryMetadata.sol";
+
+/// @title MetadataMixin
+/// @notice Mixin contract for Registry implementations to delegate metadata to an external provider
+/// @dev Inherit this contract to add metadata functionality to Registry contracts
+abstract contract MetadataMixin {
+    /// @notice The metadata provider contract
+    IRegistryMetadata public immutable metadataProvider;
```

Comment:
I think this probably needs to be mutable, with a role and a method to change it.

Reply:
I want to wait until after the new EAC stuff is in place before assigning a role. Plus, perhaps some registries will want to re-use an existing role for this. So for now I'm just going to add an internal method to update the value.